File: Assignment10/1)/Complex.java
public class Complex {
    private double real;
    private double imag;

   
    public Complex(double real, double imag) {
        this.real = real;
        this.imag = imag;
    }

    
    public Complex() {
        this(0.0, 0.0);
    }

    
    public double getReal() {
        return real;
    }

    
    public void setReal(double real) {
        this.real = real;
    }

    
    public double getImag() {
        return imag;
    }

  
    public void setImag(double imag) {
        this.imag = imag;
    }

   
    public void setValue(double real, double imag) {
        this.real = real;
        this.imag = imag;
    }

   
    public void print() {
        System.out.print(real);
        if (imag >= 0) {
            System.out.println(" + " + imag + "i");
        } else {
            System.out.println(" - " + -imag + "i");
        }
    }

   
    public boolean isReal() {
        return imag == 0.0;
    }

   
    public boolean isImaginary() {
        return real == 0.0;
    }

   
    public Complex addInto(Complex another) {
        this.real += another.real;
        this.imag += another.imag;
        return this;
    }

   
    public Complex addInto(double real, double imag) {
        this.real += real;
        this.imag += imag;
        return this;
    }

    
    public Complex addReturnNew(Complex another) {
        return new Complex(this.real + another.real, this.imag + another.imag);
    }

    
    public Complex addReturnNew(double real, double imag) {
        return new Complex(this.real + real, this.imag + imag);
    }
}

File: Assignment10/1) C++/Complex.cpp
#include "Complex.h"
#include <iostream>

using namespace std;


Complex::Complex(double real, double imag) : real(real), imag(imag) {}


double Complex::getReal() const {
    return real;
}


void Complex::setReal(double real) {
    this->real = real;
}


double Complex::getImag() const {
    return imag;
}


void Complex::setImag(double imag) {
    this->imag = imag;
}


void Complex::setValue(double real, double imag) {
    this->real = real;
    this->imag = imag;
}


void Complex::print() const {
    cout << real;
    if (imag >= 0) {
        cout << " + " << imag << "i" << endl;
    } else {
        cout << " - " << -imag << "i" << endl;
    }
}


bool Complex::isReal() const {
    return imag == 0.0;
}


bool Complex::isImaginary() const {
    return real == 0.0;
}


Complex & Complex::addInto(const Complex & another) {
    this->real += another.real;
    this->imag += another.imag;
    return *this;
}


Complex & Complex::addInto(double real, double imag) {
    this->real += real;
    this->imag += imag;
    return *this;
}


Complex Complex::addReturnNew(const Complex & another) const {
    return Complex(this->real + another.real, this->imag + another.imag);
}


Complex Complex::addReturnNew(double real, double imag) const {
    return Complex(this->real + real, this->imag + imag);
}
